example/ROS_demo/face_detector_node.cpp

```diff
@@ -0,0 +1,157 @@
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+* This file is part of the Face_Detector.Cpp library
+*
+* Author:         Jeroen Veen <ducroq>
+*
+*/
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/image.hpp"
+#include "sensor_msgs/msg/region_of_interest.hpp"
+#include "cv_bridge/cv_bridge.hpp"
+#include <opencv2/opencv.hpp>
+#include "std_msgs/msg/int32.hpp"
+#include "geometry_msgs/msg/point_stamped.hpp"
+#include "std_srvs/srv/set_bool.hpp"
+#include <face_detection.hpp>
+
+class FaceDetectorNode : public rclcpp::Node
+{
+public:
+    FaceDetectorNode() : Node("face_detector_node")
+    {
+        declare_and_get_parameters();
+        det_.load_model(CFML_FACE_DETECTOR_CPU_MODEL_PATH);
+        setup_communication();
+        detection_enabled_ = true;
+    }
+
+private:
+    void declare_and_get_parameters()
+    {
+        declare_parameter("camera_topic", "/camera/image_raw");
+        declare_parameter("face_detected_topic", "/face_detected");
+        declare_parameter("face_roi_topic", "/face_roi");
+        declare_parameter("face_landmarks_topic", "/face_landmarks");
+
+        camera_topic_ = get_parameter("camera_topic").as_string();
+        face_detected_topic_ = get_parameter("face_detected_topic").as_string();
+        face_roi_topic_ = get_parameter("face_roi_topic").as_string();
+        face_landmarks_topic_ = get_parameter("face_landmarks_topic").as_string();
+    }
+
+    void setup_communication()
+    {
+        image_sub_ = create_subscription<sensor_msgs::msg::Image>(
+            camera_topic_, rclcpp::SensorDataQoS(), 
+            std::bind(&FaceDetectorNode::imageCb, this, std::placeholders::_1));
+        
+        face_detected_pub_ = create_publisher<std_msgs::msg::Int32>(face_detected_topic_, 10);
+        face_roi_pub_ = create_publisher<sensor_msgs::msg::RegionOfInterest>(face_roi_topic_, 10);        
+        face_landmarks_pub_ = create_publisher<geometry_msgs::msg::PointStamped>(face_landmarks_topic_, 10);
+
+        toggle_service_ = create_service<std_srvs::srv::SetBool>(
+            "toggle_face_detection", 
+            std::bind(&FaceDetectorNode::toggleDetection, this, std::placeholders::_1, std::placeholders::_2));
+    }
+
+    void imageCb(const sensor_msgs::msg::Image::SharedPtr msg)
+    {
+        if (!detection_enabled_) return;
+        
+        cv_bridge::CvImagePtr cv_ptr;
+        try
+        {
+            cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
+        }
+        catch (cv_bridge::Exception& e)
+        {
+            RCLCPP_ERROR(get_logger(), "cv_bridge exception: %s", e.what());
+            return;
+        }
+
+        det_.load_image(cv_ptr->image);
+
+        publish_face_detected();
+        publish_face_roi();
+        publish_face_landmarks();
+    }
+
+    void publish_face_detected()
+    {
+        auto face_detected_msg = std_msgs::msg::Int32();
```

Comment:
I favor distinguished data types in favor of auto, as deducted types can sometimes cause bugs/be unreadable

Reply:
ok, I am going to remove all auto's and retest